File: javav2/example_code/timestream/src/main/java/com/timestream/write/DescribeTable.java
//snippet-sourcedescription:[DescribeTable.java demonstrates how to return information about a database table.]
//snippet-keyword:[AWS SDK for Java v2]
//snippet-keyword:[Code Sample]
//snippet-keyword:[Amazon Timestream]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[05-04-2022]
//snippet-sourceauthor:[scmacdon- AWS]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.timestream.write;

//snippet-start:[timestream.java2.desc_table.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.timestreamwrite.TimestreamWriteClient;
import software.amazon.awssdk.services.timestreamwrite.model.DescribeTableRequest;
import software.amazon.awssdk.services.timestreamwrite.model.DescribeTableResponse;
import software.amazon.awssdk.services.timestreamwrite.model.TimestreamWriteException;
//snippet-end:[timestream.java2.desc_table.import]

/**
 * Before running this SDK for Java (v2) code example, set up your development environment, including your credentials.
 *
 * For more information, see the following documentation:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */

public class DescribeTable {

    public static void main(String[] args) {

        final String usage = "\n" +
                "Usage: " +
                "   <dbName> <tableName>\n\n" +
                "Where:\n" +
                "   dbName - The name of the database.\n\n" +
                "   tableName - The name of the table.\n\n";

        if (args.length != 2) {
            System.out.println(usage);
            System.exit(1);
        }

        String dbName = args[0];
        String tableName = args[1];
        TimestreamWriteClient timestreamWriteClient = TimestreamWriteClient.builder()
                .region(Region.US_EAST_1)
                .build();

        describeSingleTable(timestreamWriteClient,dbName, tableName);
        timestreamWriteClient.close();
    }

    //snippet-start:[timestream.java2.desc_table.main]
    public static void describeSingleTable(TimestreamWriteClient timestreamWriteClient, String dbName, String tableName) {

        try {

            System.out.println("Describing table");
            DescribeTableRequest describeTableRequest = DescribeTableRequest.builder()
                    .databaseName(dbName)
                    .tableName(tableName)
                    .build();

            DescribeTableResponse response = timestreamWriteClient.describeTable(describeTableRequest);
            String tableId = response.table().arn();
            System.out.println("Table " + tableName + " has id " + tableId);

        } catch (TimestreamWriteException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
    //snippet-end:[timestream.java2.desc_table.main]
}

File: .kotlin_alpha/services/dynamodb/src/main/kotlin/com/kotlin/dynamodb/DescribeTable.kt
//snippet-sourcedescription:[DescribeTable.kt demonstrates how to retrieve information about an Amazon DynamoDB table.]
//snippet-keyword:[AWS SDK for Kotlin]
//snippet-keyword:[Code Sample]
//snippet-service:[Amazon DynamoDB]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[03/02/2021]
//snippet-sourceauthor:[scmacdon-aws]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.dynamodb

// snippet-start:[dynamodb.kotlin.describe_table.import]
import aws.sdk.kotlin.services.dynamodb.DynamoDbClient
import aws.sdk.kotlin.services.dynamodb.model.DescribeTableRequest
import aws.sdk.kotlin.services.dynamodb.model.DynamoDbException
import kotlin.system.exitProcess
// snippet-end:[dynamodb.kotlin.describe_table.import]

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <tableName>

    Where:
        tableName - the Amazon DynamoDB table to get information about (for example, Music3).
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val tableName = args[0]
    val ddb = DynamoDbClient{ region = "us-east-1" }
    describeDymamoDBTable(ddb, tableName);
    ddb.close()
}

// snippet-start:[dynamodb.kotlin.describe_table.main]
suspend fun describeDymamoDBTable(ddb: DynamoDbClient, tableNameVal: String?) {

        val request = DescribeTableRequest {
            tableName= tableNameVal
        }

        try {
            val tableInfo = ddb.describeTable(request)
            println("Table name ${tableInfo.table?.tableName}")
            println("Table Arn:  ${tableInfo.table?.tableArn}")
            println("Table Status: ${tableInfo.table?.tableStatus}")
            println("Item count:  ${tableInfo.table?.itemCount}")
            println("Size (bytes): ${tableInfo.table?.tableSizeBytes}")

        } catch (ex: DynamoDbException) {
            println(ex.message)
            ddb.close()
            exitProcess(0)
        }
 }
// snippet-end:[dynamodb.kotlin.describe_table.main]